app/src/main/java/org/mozilla/focus/browser/HomeTileRecyclerView.kt

```diff
@@ -97,9 +97,20 @@ class HomeTileAdapter(
         return null
     }
 
-    fun addItemToAdapter(homeTile: HomeTile) {
-        tiles.add(homeTile)
-        notifyItemInserted(tiles.lastIndex)
+    fun updateAdapterSingleInsertion(homeTiles: MutableList<HomeTile>) {
```

Comment:
This is a lot of non-obvious code - add a javadoc function comment explaining what you're trying to do and **why** you're doing it.

E.g., think about the case where you're a new hire reading this code - it would be helpful to explain why we're doing this comparison, rather than having to read through the code and then realize that it's comparing item-by-item.